### PR TITLE
Add io.github.gamingdoom.Datcord

### PR DESCRIPTION
**Pull request by gamingdoom**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

This application does use more than standard permissions. Since Datcord is based off of Firefox, I had to use similar finish args to Firefox for the same reason as Firefox used those args (although I removed a few like networkmanager dbus talk, pcsc). However, I had to add /tmp/open-in-default-browser filesystem because the open-in-default-browser script has to be in a temp directory decided before Datcord is launched that needs to be an absolute path. Home directory permission is used to store firefox profiles to store discord passwords.

### CODE REVIEW

File: default128.png
Comment (hfiguiere) [inline default128.png:1]: since it doesn't come from upstream, keep only the biggest icon.

File: io.github.gamingdoom.Datcord.desktop [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[Desktop Entry]
+Name=Datcord
+Exec=/app/lib/datcord/launch-datcord
```
Comment (hfiguiere) [inline io.github.gamingdoom.Datcord.desktop:3]: this should NO be an absolute path.

File: io.github.gamingdoom.Datcord.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
+app-id: io.github.gamingdoom.Datcord
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+base: org.mozilla.firefox.BaseApp
+base-version: '23.08'
+command: /app/lib/datcord/launch-datcord
```
Comment (hfiguiere) [inline io.github.gamingdoom.Datcord.yml:7]: this shold not be an absolute path.

File: io.github.gamingdoom.Datcord.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+rename-icon: io.github.gamingdoom.Datcord
```
Comment (hfiguiere) [inline io.github.gamingdoom.Datcord.yml:8]: rename to the same thing?

File: io.github.gamingdoom.Datcord.yml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    - --filesystem=xdg-download:rw                            
+    - --filesystem=/run/.heim_org.h5l.kcm-socket              
+    - --filesystem=/tmp/open-in-default-browser                                       
+    - --device=all                                            
+    - --talk-name=org.freedesktop.FileManager1                
+    - --talk-name=org.a11y.Bus                                
+    - --talk-name=org.gnome.SessionManager                    
+    - --talk-name=org.freedesktop.ScreenSaver                 
+    - --talk-name=org.freedesktop.Notifications               
+    - --command=/app/lib/datcord/launch-datcord
```
Comment (hfiguiere) [inline io.github.gamingdoom.Datcord.yml:36]:
```suggestion
```

this is invalid.

File: io.github.gamingdoom.Datcord.yml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    - --filesystem=home:rw
+    - --env=GTK_THEME_VARIANT=dark
+    - --device=dri
```
Comment (hfiguiere) [inline io.github.gamingdoom.Datcord.yml:39]:
```suggestion
```

you already have devie all.

File: io.github.gamingdoom.Datcord.yml [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    - --filesystem=/tmp/open-in-default-browser                                       
+    - --device=all                                            
+    - --talk-name=org.freedesktop.FileManager1                
+    - --talk-name=org.a11y.Bus                                
+    - --talk-name=org.gnome.SessionManager                    
+    - --talk-name=org.freedesktop.ScreenSaver                 
+    - --talk-name=org.freedesktop.Notifications               
+    - --command=/app/lib/datcord/launch-datcord
+    - --filesystem=home:rw
+    - --env=GTK_THEME_VARIANT=dark
```
Comment (hfiguiere) [inline io.github.gamingdoom.Datcord.yml:38]:
```suggestion
```

No. you don't get to choose the theme this way.

File: flathub.json [hunk 7]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: this is a disappointment.

File: io.github.gamingdoom.Datcord.desktop (hunk 1 above)
Comment (gamingdoom) [reply io.github.gamingdoom.Datcord.desktop:3]: What do you recommend to change it to? Will /app/bin work?

File: io.github.gamingdoom.Datcord.yml [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    - --filesystem=/run/.heim_org.h5l.kcm-socket              
+    - --filesystem=/tmp/open-in-default-browser                                       
+    - --device=all                                            
+    - --talk-name=org.freedesktop.FileManager1                
+    - --talk-name=org.a11y.Bus                                
+    - --talk-name=org.gnome.SessionManager                    
+    - --talk-name=org.freedesktop.ScreenSaver                 
+    - --talk-name=org.freedesktop.Notifications               
+    - --command=/app/lib/datcord/launch-datcord
+    - --filesystem=home:rw
```
Comment (hfiguiere) [inline io.github.gamingdoom.Datcord.yml:37]: why do you need home access for a discord client?

File: screenshot1.png
Comment (hfiguiere) [inline screenshot1.png:1]: why do you put screenshots here? they are not referenced anywhere.

File: flathub.json (hunk 7 above)
Comment (gamingdoom) [reply flathub.json:2]: The manifest is using the binary release of my application and I don't currently have the resources to compile for arm.

File: io.github.gamingdoom.Datcord.yml [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+add-build-extensions:
+  org.freedesktop.Platform.ffmpeg-full:
+    directory: lib/ffmpeg
+    add-ld-path: .
+    version: '23.08'
+    autodownload: true
+    autodelete: false
+
+finish-args:
+    - --allow=devel
```
Comment (hfiguiere) [inline io.github.gamingdoom.Datcord.yml:18]: really?

File: io.github.gamingdoom.Datcord.yml [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+      - install -Dm 644 -t /app/share/applications io.github.gamingdoom.Datcord.desktop
+      - install -Dm 644 -t /app/share/metainfo io.github.gamingdoom.Datcord.metainfo.xml
+
+      - install -Dm 644 default256.png /app/share/icons/hicolor/256x256/apps/io.github.gamingdoom.Datcord.png
+      - install -Dm 644 default128.png /app/share/icons/hicolor/128x128/apps/io.github.gamingdoom.Datcord.png
+
+    sources:
+      - type: archive
+        url: https://github.com/gamingdoom/datcord/releases/latest/download/datcord-linux-x86_64.tar.bz2
```
Comment (hfiguiere) [inline io.github.gamingdoom.Datcord.yml:62]: no. you need a tag or a specific commit here.

File: flathub.json (hunk 7 above)
Comment (hfiguiere) [reply flathub.json:2]: why don't you let flathub build it. like it should be.

File: io.github.gamingdoom.Datcord.yml (hunk 8 above)
Comment (gamingdoom) [reply io.github.gamingdoom.Datcord.yml:37]: Home access is to create a .datcord directory to save firefox profiles containing the password for the discord account.

Comment (hfiguiere) [reply io.github.gamingdoom.Datcord.yml:37]: You have `persist` above.

Better would be that your app follow the XDG spec.

**Comment by gamingdoom** [comment]
I have fixed all of the issues pointed out except for the arm compatibility.

**Comment by gamingdoom** [comment]
Turns out, after more testing that home directory access isn't necessary to store the profiles, so I have removed it.

File: io.github.gamingdoom.Datcord.yml (hunk 8 above)
Comment (gamingdoom) [reply io.github.gamingdoom.Datcord.yml]: Upon further inspection, it works without home access so I have removed it.

File: flathub.json (hunk 7 above)
Comment (gamingdoom) [reply flathub.json:2]: I am working on having the manifest build it from source so it should build on arm.

**Comment by gamingdoom** [comment]
The manifest now compiles Datcord from source, allowing for arm support.

File: flathub.json (hunk 7 above)
Comment (gamingdoom) [reply flathub.json]: The manifest now builds it from source, allowing for arm support.

File: default256.png
Comment (barthalion) [inline default256.png:1]: You should consider changing the logo to something that isn't just a recoloured Discord logo.

File: io.github.gamingdoom.Datcord.yml [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+    - type: file
+      path: io.github.gamingdoom.Datcord.desktop
+    
+    - type: file
+      path: io.github.gamingdoom.Datcord.metainfo.xml
+
+    - type: file
+      path: default256.png
+
+  # - name: datcord
```
Comment (barthalion) [inline io.github.gamingdoom.Datcord.yml:136]: Please remove the commented out section.

File: io.github.gamingdoom.Datcord.yml [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+    - --filesystem=/run/.heim_org.h5l.kcm-socket              
+    - --filesystem=/tmp/open-in-default-browser                                       
+    - --device=all                                            
+    - --talk-name=org.freedesktop.FileManager1                
+    - --talk-name=org.a11y.Bus                                
+    - --talk-name=org.gnome.SessionManager                    
+    - --talk-name=org.freedesktop.ScreenSaver                 
+    - --talk-name=org.freedesktop.Notifications               
+    - --filesystem=~/.config/dconf:ro
+    - --env=DCONF_USER_CONFIG_DIR=.config/dconf
```
Comment (barthalion) [inline io.github.gamingdoom.Datcord.yml:41]: What are these two needed for?

File: io.github.gamingdoom.Datcord.yml [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+    - --share=ipc                                             
+    - --share=network                                         
+    - --env=GTK_PATH=/app/lib/gtkmodules                      
+    - --socket=pulseaudio                                     
+    - --socket=wayland                                        
+    - --socket=fallback-x11                                            
+    - --require-version=0.11.1                                
+    - --persist=.datcord                                      
+    - --filesystem=xdg-download:rw                            
+    - --filesystem=/run/.heim_org.h5l.kcm-socket
```
Comment (barthalion) [inline io.github.gamingdoom.Datcord.yml:32]: I can imagine you are not signing in to Discord with sssd.

File: io.github.gamingdoom.Datcord.yml [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+    autodelete: false
+
+finish-args:
+    - --share=ipc                                             
+    - --share=network                                         
+    - --env=GTK_PATH=/app/lib/gtkmodules                      
+    - --socket=pulseaudio                                     
+    - --socket=wayland                                        
+    - --socket=fallback-x11                                            
+    - --require-version=0.11.1
```
Comment (barthalion) [inline io.github.gamingdoom.Datcord.yml:29]: 0.11 is from around 2018, it's fine not to define it at this point.

**Comment by barthalion** [comment]
bot, build io.github.gamingdoom.Datcord

**Comment by barthalion** [comment]
There's no Firefox baseapp for aarch64. Please copy the modules from its manifest and don't use it directly, it's not meant for anything but Firefox and Thunderbird.

**Comment by gamingdoom** [comment]
I have implemented your suggestions with cleaning up the manifest and not using the firefox base app. I have copied the firefox base app's modules. However, I don't currently have a different logo, so I have made it grab the logo from upstream.

**Comment by gamingdoom** [comment]
bot, build io.github.gamingdoom.Datcord

**Comment by barthalion** [comment]
bot, build io.github.gamingdoom.Datcord

**Comment by barthalion** [comment]
Please make sure to explore using an alternative logo.

/merge

**Pull request closed without merging**